### PR TITLE
#17 feature/effective-config

### PR DESCRIPTION
**Pull request by Valvon**
Add Configuration Service that listens to configuration files and provides the effective configuration (lazy).
Add ToolWindow for the Effective Configuration

### CODE REVIEW

File: src/main/kotlin/org/jqassistant/tooling/intellij/plugin/data/config/CommandLineTool.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import java.io.BufferedReader
+import java.io.File
+import java.io.InputStreamReader
+
+// TODO - the ProcessBuilder running the maven goal will be replaced by jqa api. This will NOT be the permanent solution and file location
+
+class CommandLineTool {
+    fun runMavenGoal(goal: String, directory: File): String {
+        var result = ""
+        val mvn = getMavenCmdPath()?.path ?: ""
```
[the remark this answers is not in the document]
Comment (Valvon) [reply src/main/kotlin/org/jqassistant/tooling/intellij/plugin/data/config/CommandLineTool.kt:12]: agreed

File: src/main/kotlin/org/jqassistant/tooling/intellij/plugin/data/config/JqaCliRunner.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    private val jqa = "jqassistant:effective-configuration"
+    private val jqaCLI = Main()
+    private val task = EffectiveConfigurationTask()
+
+
+    init {
+
+        val config = runEffectiveConfigurationTask()
+        println(config)
```
[the remark this answers is not in the document]
Comment (Valvon) [reply src/main/kotlin/org/jqassistant/tooling/intellij/plugin/data/config/JqaCliRunner.kt:18]: Noted, but seems like this is a leftover from development and line should be deleted

File: src/main/kotlin/org/jqassistant/tooling/intellij/plugin/editor/effectiveConfig/LoadingPanel.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import java.awt.BorderLayout
+import javax.swing.ImageIcon
+import javax.swing.JLabel
+import javax.swing.JPanel
+
+class LoadingPanel : JPanel() {
+    init {
+        layout = BorderLayout()
+        val loadingIcon = ImageIcon(javaClass.getResource("/icons/loading.gif"))
```
Comment (A-K-O-R-A) [inline src/main/kotlin/org/jqassistant/tooling/intellij/plugin/editor/effectiveConfig/LoadingPanel.kt]:
```suggestion
        val loadingIcon = AnimatedIcon.Default()
```
Using the IntelliJ provided loading Icon works better with high DPI screens and is also used in the IntelliJ codebase

File: src/main/kotlin/org/jqassistant/tooling/intellij/plugin/editor/config/toolwindow/LoadingPanel.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import java.awt.BorderLayout
+import javax.swing.ImageIcon
+import javax.swing.JLabel
+import javax.swing.JPanel
+
+class LoadingPanel : JPanel() {
+    init {
+        layout = BorderLayout()
+        val loadingIcon = ImageIcon(javaClass.getResource("/icons/loading.gif"))
```
Comment (A-K-O-R-A) [inline src/main/kotlin/org/jqassistant/tooling/intellij/plugin/editor/config/toolwindow/LoadingPanel.kt:11]:
```suggestion
        val loadingIcon = AnimatedIcon.Default()
```
Using the AnimatedIcon class works better on HighDPI screens and is commonly used in the intellij codebase

**Review by A-K-O-R-A — CHANGES_REQUESTED** [review]

**Review by A-K-O-R-A — APPROVED** [review]

**Pull request merged**